pkg/client/apiutil/apimachinery.go

```diff
@@ -64,7 +64,7 @@ func AddToProtobufScheme(addToScheme func(*runtime.Scheme) error) error {
 // information fetched by a new client with the given config.
 func NewDiscoveryRESTMapper(c *rest.Config, httpClient *http.Client) (meta.RESTMapper, error) {
 	if httpClient == nil {
-		panic("httpClient must not be nil")
+		return nil, fmt.Errorf("httpClient must not be nil")
```

Comment:
Maybe include the info into the error how ppl can construct one from a rest config?